src/test/java/com/box/sdk/MetadataTemplateTest.java

```diff
@@ -53,6 +53,36 @@ public String getJSON() {
         MetadataTemplate.getMetadataTemplate(api, "properties", "global", "displayName", "hidden");
     }
 
+    @Test
+    @Category(UnitTest.class)
+    public void testDeleteMetadataTemplateSendsCorrectRequest() {
+        BoxAPIConnection api = new BoxAPIConnection("");
+        api.setRequestInterceptor(new RequestInterceptor() {
+            @Override
+            public BoxAPIResponse onRequest(BoxAPIRequest request) {
+                Assert.assertEquals(
+                        "https://api.box.com/2.0/metadata_templates/enterprise/testtemplate/schema",
+                        request.getUrl().toString());
+                return new BoxAPIResponse() {
+                    public String getJSON() {
+                        return "{\"status\": \"204\"}";
+                    }
+                };
+            }
+        });
+        MetadataTemplate.deleteMetadataTemplate(api, "enterprise", "testtemplate");
+    }
+
+    @Test
+    @Category(IntegrationTest.class)
+    public void testDeleteMetadataTemplateSucceeds() {
+        String scope = "";
+        String template = "";
+
+        BoxAPIConnection api = new BoxAPIConnection(TestConfig.getAccessToken());
+        MetadataTemplate.deleteMetadataTemplate(api, scope, template);
```

Comment:
This test doesn't seem to really verify that the method does what it says.  Can we instead create a new template, then delete it, and then assert that it no longer exists?